common.tasks

```diff
@@ -4,28 +4,40 @@
   <UsingTask TaskName="Zip" TaskFactory="CodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.v4.0.dll">
     <ParameterGroup>
       <InputFileNames ParameterType="Microsoft.Build.Framework.ITaskItem[]" Required="true" />
+      <InputBaseDirectory ParameterType="System.String" Required="false" />
       <OutputFileName ParameterType="System.String" Required="true" />
       <OverwriteExistingFile ParameterType="System.Boolean" Required="false" />
-	</ParameterGroup>
+    </ParameterGroup>
     <Task>
       <Reference Include="System.IO.Compression" />
       <Using Namespace="System.IO.Compression" />
       <Code Type="Fragment" Language="cs">
-      <![CDATA[        
+      <![CDATA[
+        Log.LogMessage(MessageImportance.High, "=== Zipping files ===");
+        
         var fileMode = OverwriteExistingFile ? FileMode.Create : FileMode.CreateNew;
-        using (var archive = new ZipArchive(new FileStream(OutputFileName, fileMode), ZipArchiveMode.Create))
+        var baseDir  = string.IsNullOrWhiteSpace(InputBaseDirectory) ? string.Empty : Path.GetFullPath(InputBaseDirectory);
+        var filesToZip = InputFileNames.Select(f => Path.GetFullPath(f.ItemSpec)).Where(n => n.StartsWith(baseDir));
+        
+        using (var archive = new ZipArchive(new FileStream(OutputFileName, fileMode), ZipArchiveMode.Create, leaveOpen: false))
         {
-            foreach (var inputFileName in InputFileNames.Select(f => f.ItemSpec))
+          foreach (var inputFileName in filesToZip)
+          {
+            var entryName = baseDir.Length == 0
+                          ? Path.GetFileName(inputFileName) 
+                          : inputFileName.Substring(baseDir.Length + 1);
+            
+            Log.LogMessage(MessageImportance.High, "Zipping '" + entryName + "' from " + inputFileName);
```

Comment:
Not sure why this is useful. This will just clutter the logs especially where there is a lot of files to zip.